tests/api/test_recurring_api.py

```diff
@@ -1,110 +1,89 @@
+        paypalPaymentPreferences.setup_fee = PayPalPaymentPreferencesSetUpFee()
+        paypalPaymentPreferences.setup_fee.value = 20
+        paypalPaymentPreferences.setup_fee.currency_code = "USD"
+        paypalPaymentPreferences.setup_fee_failure_action = "CONTINUE"
+        paypalPaymentPreferences.Payment_failure_threshold = 3
+
+        # Declare PayPal Taxes Object
+        paypalTaxes = PayPalTaxes()
+        paypalTaxes.percentage = "10"
+        paypalTaxes.inclusive = True
+
+        # Declare PayPal Product Schema Object
+        payPalProductSchema = PayPalProductSchema()
+        payPalProductSchema.type = "SERVICE"
+        payPalProductSchema.category = "SOFTWARE"
 
-        # Perform the API call through the SDK function
-        response = self.client.auth.auth_capture(params)
-        data = response.body
-        # Test response code
-        self.assertEqual(data['ret_code'], '000100')
-
-    # Make Auth Detail Query request.
-    def test_auth_detail_query(self):
-        # Parameters for the API call
-        params = {
-            'outAuthInfoNo': '298217806906830305',
-            'outAuthDetailNo': '0000'
-        }
-
-        # Perform the API call through the SDK function
-        response = self.client.auth.auth_detail_query(params)
-        data = response.body
-        # Test response code
-        self.assertEqual(data['ret_code'], '000100')
-
-    # Make Auth Unfreeze request.
-    def test_auth_unfreeze(self):
-        # Parameters for the API call
-        params = {
-            'unfreezeAmount': '0.01',
-            'outAuthInfoNo': '298217806906830305',
-            'currency': 'USD',
-            'settleCurrency': 'USD',
-            'outAuthDetailNo': '0000'
-        }
-
-        # Perform the API call through the SDK function
-        response = self.client.auth.auth_unfreeze(params)
-        data = response.body
-        # Test response code
-        self.assertEqual(data['ret_code'], '000100')
-
-    # Make Auth Freeze request.
-    def test_auth_freeze(self):
-        # Parameters for the API call
-        params = {
-            'amount': '0.01',
-            'outAuthInfoNo': '298217806906830305',
-            'currency': 'USD',
-            'settleCurrency': 'USD',
-            'outAuthDetailNo': '0000',
-            'vendor': 'alipay'
-        }
-
-        # Perform the API call through the SDK function
-        response = self.client.auth.auth_freeze(params)
-        data = response.body
-        # Test response code
-        self.assertEqual(data['ret_code'], '000100')
-
-    # Make Auth Voucher Create request.
-    def test_auth_voucher_create(self):
-        # Parameters for the API call
         params = {
-            'amount': '0.01',
-            'outAuthInfoNo': '298217806906830305',
-            'outAuthDetailNo': '0000',
-            'vendor': 'alipay'
+            "clientId": "AXV4uwyZ5WY9zpYs7zaLrnPcHX4s9AA0VdxEX2mo23UTqmVl_aH7V_p0Nguv5sdIB2u3osE40hqIbE7U",
```

Comment:
Should we even hardcode this? Why even put the `clientId` and `secret` here?

Reply:
oh right, let me hide it

Reply:
are we using the real ENV to test? if it's just unit test, can we mock it? I am not really familiar with python though.

Reply:
you mean the clientId and secret id? i am just using the sandbox credentials

Reply:
+1 with mocking it instead of using creds (sandbox or not)